### PR TITLE
Add address validation

### PR DESCRIPTION
**Pull request by lulf**

### CODE REVIEW

File: api-model/src/main/java/io/enmasse/address/model/Address.java [hunk 1]
```diff
@@ -130,16 +132,24 @@ public void validate() {
     private void validateName(String name) {
         String[] components = name.split("\\.");
         if (components.length < 2) {
-            throw new IllegalArgumentException("Address name must be on the form addressSpace.addressName");
+            throw new AddressValidationFailedException("Address name must be on the form addressSpace.addressName");
```
Comment (RH-sdavey) [inline api-model/src/main/java/io/enmasse/address/model/Address.java]: must be in the form

**Comment by lulf** [comment]
Is it really not allowed to create addresses with '/' ? @grs  I think this is already used by Bosch (CC @calohmn ).

**Comment by grs** [comment]
@lulf you can relax it if it is too strict. I would check how artemis behaves with a queue that has a / in it though, may be fine. (For anycast it certainly would not cause any problem)

**Comment by k-wall** [comment]
What is it that motivates us to make this change?

Do we have a definition somewhere  of what constitutes a valid address in EnMasse?  I think this should appear in user-documentation too. 

I tried creating addresses and queue in Artemis containing the characters excluded by the PR.  It didn't seem to cause an issue.  I couldn't find any documentation.

**Comment by lulf** [comment]
@k-wall It was motivated by the referenced issue #2179 where the console is restricting the allowed characters for addresses. I think we should perhaps leave it out and do more experimentation before we end up restricting it more than necessary.

**Comment by RH-sdavey** [comment]
+1 for proper documentation of what is a valid/invalid address if possible, 
also, I noticed there were differences between what addresses were possible to be made through API server and through console, where they would be accepted in one and not the other, so maybe some standardization on that?

**Comment by lulf** [comment]
Have to be rethought. Also underlying model changes in #2202

**Pull request closed without merging**